test/libsolidity/smtCheckerTests/simple/static_array.sol

```diff
@@ -0,0 +1,8 @@
+pragma experimental SMTChecker;
+contract C
+{
+	// Used to crash because Literal had no type
+	int[3] d;
```

Comment:
Why was this only crashing in SMT?

Reply:
Because the types of user defined type names are resolved in the referenceresolver. The type checker then later does not descend into such AST items anymore. So it is only the reference resolver (who does it itself) and the smt checker that take a look at this.